src/compiler/checker.ts

```diff
@@ -594,10 +594,16 @@ namespace ts {
                 getRestTypeOfSignature,
                 getTypePredicateOfSignature,
                 getReturnTypeOfSignature,
-                getBaseTypes,
+                type => {
+                    Debug.assert(getObjectFlags(type) & ObjectFlags.ClassOrInterface);
+                    return getBaseTypes(type as InterfaceType);
+                },
                 resolveStructuredTypeMembers,
                 getTypeOfSymbol,
-                getResolvedSymbol,
+                node => {
+                    Debug.assert(isIdentifier(node));

```

Comment:
Probably should be `Debug.assertNode(node, isIdentifier)`, but I know that one's not always checked thanks to some logging level magic.